azurerm/internal/services/datafactory/data_factory_dataset_binary_resource.go

```diff
@@ -0,0 +1,400 @@
+				MaxItems: 1,
+				Optional: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						// TarGZip, GZip, ZipDeflate
+						"level": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"Optimal",
+								"Fastest",
+							}, false),
+						},
+						// SFTP Specific field
+						"type": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(datafactory.TypeBasicDatasetCompressionTypeBZip2),
+								string(datafactory.TypeBasicDatasetCompressionTypeDeflate),
+								string(datafactory.TypeBasicDatasetCompressionTypeGZip),
+								string(datafactory.TypeBasicDatasetCompressionTypeTar),
+								string(datafactory.TypeBasicDatasetCompressionTypeTarGZip),
+								string(datafactory.TypeBasicDatasetCompressionTypeZipDeflate),
+							}, false),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceDataFactoryDatasetBinaryCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.DatasetClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id := parse.NewDataSetID(subscriptionId, d.Get("resource_group_name").(string), d.Get("data_factory_name").(string), d.Get("name").(string))
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id.ResourceGroup, id.FactoryName, id.Name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing Data Factory Dataset Binary %q (Data Factory %q / Resource Group %q): %s", id.Name, id.FactoryName, id.ResourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_factory_dataset_binary", *existing.ID)
+		}
+	}
+
+	location := expandDataFactoryDatasetLocation(d)
+	if location == nil {
+		return fmt.Errorf("one of `http_server_location`, `azure_blob_storage_location` or `sftp_server_location`, must be specified to create a DataFactory Binary Dataset")
+	}
+
+	binaryDatasetProperties := datafactory.BinaryDatasetTypeProperties{
+		Location: location,
+	}
+
+	if _, ok := d.GetOk("compression"); ok {
+		binaryDatasetProperties.Compression = expandDataFactoryDatasetCompression(d)
+	}
+
+	linkedServiceName := d.Get("linked_service_name").(string)
+	linkedServiceType := "LinkedServiceReference"
+	linkedService := &datafactory.LinkedServiceReference{
+		ReferenceName: &linkedServiceName,
+		Type:          &linkedServiceType,
+	}
+
+	description := d.Get("description").(string)
+	binaryTableset := datafactory.BinaryDataset{
+		BinaryDatasetTypeProperties: &binaryDatasetProperties,
+		LinkedServiceName:           linkedService,
+		Description:                 &description,
+	}
```

Comment:
could we simplify this some?
```suggestion

	binaryTableset := datafactory.BinaryDataset{
		BinaryDatasetTypeProperties: &binaryDatasetProperties,
		Description:                 utils.String(d.Get("description").(string)),
		LinkedServiceName:           datafactory.LinkedServiceReference{
		    ReferenceName:           utils.string(d.Get("linked_service_name").(string)),
		    Type:                    utils.string("LinkedServiceReference"),
	    },
	}
```

Reply:
Thanks!